tensorflow_probability/examples/cifar10_bnn.py

```diff
@@ -0,0 +1,246 @@
+                         "bayesian_neural_network/"),
+    help="Directory to put the model's fit.")
+flags.DEFINE_integer("eval_freq",
+                     default=400,
+                     help="Frequency at which to validate the model.")
+flags.DEFINE_integer("num_monte_carlo",
+                     default=50,
+                     help="Network draws to compute predictive probabilities.")
+flags.DEFINE_string("arch",
+                  default="vgg",
+                  help="Network architecture to use.")
+flags.DEFINE_integer("prior_qmean",
+                     default=-9,
+                     help="Initial posterior mean of the log variance for q(w)")
+flags.DEFINE_float("variance_threshold",
+                     default=0.2,
+                     help="Posterior variance threshold constraint. Log var <= log(threshold).")
+flags.DEFINE_integer("kl_annealing",
+                     default=100,
+                     help="Epochs to anneal the KL term (anneals from 0 to 1)")
+
+FLAGS = flags.FLAGS
+
+def build_input_pipeline(x_train, x_test, y_train, y_test, 
+                         batch_size, valid_size):
+  """Build an Iterator switching between train and heldout data."""
+
+  x_train = x_train.astype('float32')
+  x_test = x_test.astype('float32')
+
+  x_train /= 255
+  x_test /= 255
+
+  y_train = y_train.flatten()
+  y_test = y_test.flatten()
+
+  print('x_train shape:' + str(x_train.shape))
+  print(str(x_train.shape[0]) + ' train samples')
+  print(str(x_test.shape[0]) + ' test samples')
+
+  # Build an iterator over training batches.
+  training_dataset = tf.data.Dataset.from_tensor_slices(
+      (x_train, np.int32(y_train)))
+  training_batches = training_dataset.shuffle(50000, 
+                                              reshuffle_each_iteration=True).repeat().batch(batch_size)
+  training_iterator = training_batches.make_one_shot_iterator()
+
+  # Build a iterator over the heldout set with batch_size=heldout_size,
+  # i.e., return the entire heldout set as a constant.
+  heldout_dataset = tf.data.Dataset.from_tensor_slices(
+      (x_test, np.int32(y_test)))
+  heldout_batches = heldout_dataset.repeat().batch(valid_size)
+  heldout_iterator = heldout_batches.make_one_shot_iterator()
+
+  # Combine these into a feedable iterator that can switch between training
+  # and validation inputs.
+  handle = tf.placeholder(tf.string, shape=[])
+  feedable_iterator = tf.data.Iterator.from_string_handle(
+      handle, training_batches.output_types, training_batches.output_shapes)
+  images, labels = feedable_iterator.get_next()
+
+  return images, labels, handle, training_iterator, heldout_iterator
+
+
+def main(argv):
+  del argv  # unused
+  if tf.gfile.Exists(FLAGS.model_dir):
+    tf.logging.warning(
+        "Warning: deleting old log directory at {}".format(FLAGS.model_dir))
+    tf.gfile.DeleteRecursively(FLAGS.model_dir)
+  tf.gfile.MakeDirs(FLAGS.model_dir)
+
+  # Load training and testing data
```

Comment:
I recommend removing comments like this, "Set training steps" below, etc. This puts more pressure on the code to be self-documenting. At the moment, it seems like it is.